agent/command/s3_copy.go

```diff
@@ -346,8 +346,9 @@ func (c *s3copy) attachFiles(ctx context.Context, comm client.Communicator,
 	}
 	logger.Execution().Infof("Attaching file '%s'.", displayName)
 	file := artifact.File{
-		Name: displayName,
-		Link: fileLink,
+		Name:        displayName,
+		Link:        fileLink,
+		ContentType: c.ContentType,
```

Comment:
I would prefer to also test this, but it seems like the s3Copy.copy tests are kind of nonexistent... 🤷

Reply:
We have https://jira.mongodb.org/browse/EVG-19764 also

Reply:
Ah, I didn't see that. Not sure how easy it is to actually deprecate without somehow giving warning in advance, because I see multiple projects in `evergreen all-configs` using it.

Reply:
I'm going to not add this functionality to s3copy if the plan is to eventually deprecate it.

Reply:
What projects do you see Kim? I actually don't see that but my configs dir could be out of date. But yeah deprecation could take a minute / might need to be axed if it's not just server projects like Brian mentioned in the linked build ticket

Reply:
@khelif96 what I meant by this thread is that I think we _should_ support content type for s3 copy since we aren't sure that removing it is going to be feasible.

Reply:
I took a look at this I would probably push back on supporting it for s3Copy.Copy. It is used .0004% of the time compared to other S3 operations. I don't think it would be worth the effort to implement given it will likely not be used at all.
https://ui.honeycomb.io/mongodb-4b/environments/production/datasets/evergreen-agent/result/fspuN6naiEn

Reply:
I did a little more research into the implementation of `s3Copy.Copy` it looks it relies on an [s3 method](https://github.com/evergreen-ci/pail/blob/74d684cc93493d325861a2d0777b519eced027ea/s3_bucket.go#L998) to just move the files between s3 buckets and does not actually ever download the file. Caching the mime type for this would require us to perform a [`GetHeadObject` ](https://github.com/evergreen-ci/pail/blob/74d684cc93493d325861a2d0777b519eced027ea/s3_bucket.go#L1387) operation for each file we wish to copy to get its ContentType and then save it. Given such low usage of the method, I don't think it's worth the additional complexity required to implement it. We can reinvestigate if people start asking for it. Although [most usage I found](https://github.com/search?q=%22s3copy.copy%22+language:YAML+owner:evergreen-ci+OR+owner:mongodb+OR+owner:10gen+&type=code) seems to only use it to move archived files around so I don't think people will notice.

Reply:
Wow honeycomb to the rescue. That works for me, we should make a note to include documentation for this or include it in the mouse over for why the button isn't available

Reply:
Done I created a [ticket](https://jira.mongodb.org/browse/EVG-20851) to write some docs for this feature and I'll make a note of it there.